### DISCUSSION TITLE
Allow custom invite link

### DISCUSSION THREAD

**Discussion by ghost** [directus/directus#3860] [Feature Requests]
Hello,

I can see there are API endpoints for `/users/invite` and `/users/invite/accept` but is there any way when calling the `/users/invite` to pass a custom URL for the email to use rather than just sending the normal admin URL as I need two invite pages that go to two different invite accept pages. My admin app is hidden so people can't find it and currently I am just sending every person that registers on my site a link to my admin area. (I am aware if they try logging in it says no app access it is just a bit unprofessional on my part)

**Comment by andriusign** [comment]
Hello. I would like to add to @Kingtona123 that indeed this functionality is very important. I want my web users to be able to enter email details (initial registration, first verifying email) on my Front-End site from where I call /users/invite. Custom link is required for security measures, which would direct users from their email to my custom Front-End page where the user would finish the registration. I also need users to be able to verify their emails not on Directus Back-End Admin (because /api/admin will be disabled for public and only certain IPs allowed for admins) but on my Front-End page.
This also depends on https://github.com/directus/directus/issues/3597

**Reply by benhaynes** [reply]
As a reminder, work that is critical (to external companies) can always be sponsored to expedite... otherwise, it will be processed within our triage workflow (after bugs, other sponsored work, and higher priority features).

https://docs.directus.io/getting-started/support.html#sponsored-work

**Reply by andriusign** [reply]
This looks similar to [resetting password](https://docs.directus.io/reference/api/rest/authentication/#request-password-reset) where you can pass "reset_url" which is then passed to email.

**Reply by andriusign** [reply]
I had some time to make a small POC which should be enough to solve this issue. Please have a look at https://github.com/andriusign/directus/commit/386f3fc10bd22bdea70bb9dba3130c9bbd639b89

**Reply by rijkvanzanten** [reply]
Feel free to open a PR for this 🙂

**Comment by rijkvanzanten** [comment]
This was implemented by @andriusign in https://github.com/directus/directus/pull/3972! ✨
